changed.d

```diff
@@ -142,56 +157,222 @@ string getChangeLog(string revRange)
+{
+    // write the overview titles
+    w.formattedWrite("$(BUGSTITLE %s,\n\n", headline);
+    scope(exit) w.put("\n)\n\n");
+    foreach(change; changes)
+    {
+        w.formattedWrite("$(LI $(RELATIVE_LINK2 %s,%s))\n", change.basename, change.title.escapeParens);
+    }
+}
+/**
+Writes the long description of the manually listed changes in the ddoc format as list.
+
+Params:
+    changes = parsed InputRange of changelog information
+    w = Output range to use
+*/
+void writeTextChangesBody(Entries, Writer)(Entries changes, Writer w, string headline)
+    if (isInputRange!Entries && isOutputRange!(Writer, string))
+{
+    w.formattedWrite("$(BUGSTITLE %s,\n\n", headline);
+    scope(exit) w.put("\n)\n\n");
+    foreach(change; changes)
+    {
+        w.formattedWrite("$(LI $(LNAME2 %s,%s)\n", change.basename, change.title.escapeParens);
+        scope(exit) w.put(")\n");
+        w.formattedWrite("    $(P %s  )\n", change.description.escapeParens);
+    }
+}
+
+/**
+Writes the fixed issued from Bugzilla in the ddoc format as a single list.
+
+Params:
+    changes = parsed InputRange of changelog information
+    w = Output range to use
+*/
+void writeBugzillaChanges(Entries, Writer)(Entries entries, Writer w)
+    if (isOutputRange!(Writer, string))
+{
+    immutable components = ["DMD Compiler", "Phobos", "Druntime", "dlang.org", "Optlink", "Tools", "Installer"];
+    immutable bugtypes = ["regressions", "bugs", "enhancements"];
 
     foreach (component; components)
-    if (auto comp = component in entries)
     {
-        foreach (bugtype; bugtypes)
-        if (auto bugs = bugtype in *comp)
+        if (auto comp = component in entries)
         {
-            result ~= format("$(BUGSTITLE %s %s,\n\n", component, bugtype);
-            foreach (bug; sort!"a.id < b.id"(*bugs))
+            foreach (bugtype; bugtypes)
+            if (auto bugs = bugtype in *comp)
             {
-                result ~= format("$(LI $(BUGZILLA %s): %s)\n",
-                              bug.id, bug.summary.escapeParens());
+                w.formattedWrite("$(BUGSTITLE %s %s,\n\n", component, bugtype);
+                foreach (bug; sort!"a.id < b.id"(*bugs))
+                {
+                    w.formattedWrite("$(LI $(BUGZILLA %s): %s)\n",
+                                        bug.id, bug.summary.escapeParens());
+                }
+                w.put(")\n");
             }
-            result ~= ")\n";
         }
     }
-
-    return result.data;
 }
 
 int main(string[] args)
 {
-    if (args.length != 2)
+    auto outputFile = "./changelog.dd";
+    auto nextVersionString = "LATEST";
+    auto nextVersionDate = "September 19, 2016";
+    string previousVersion = "Previous version";
+    string revRange;
+
+    auto helpInformation = getopt(
+        args,
+        std.getopt.config.passThrough,
+        "output|o", &outputFile,
+        "date", &nextVersionDate,
+        "version", &nextVersionString,
```

Comment:
@MartinNowak is passing a CLI flag okay for you?
(Otherwise the new version is only written once in the file - at the `Macro` section)